plugins/module_utils/inventory.py

```diff
@@ -0,0 +1,176 @@
+# Copyright: (c) 2022, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+try:
+    import boto3
+    import botocore
+except ImportError:
+    pass  # will be captured by imported HAS_BOTO3
+from abc import ABCMeta, abstractmethod
+
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible.module_utils._text import to_native
+
+
+class AWSInventoryFailure(Exception):
+    pass
+
+
+class AnsibleAWSInventory(metaclass=ABCMeta):
+
+    def __init__(self):
+
+        self.boto_profile = None
+        self.aws_secret_access_key = None
+        self.aws_access_key_id = None
+        self.aws_security_token = None
+        self.iam_role_arn = None
+
+    @abstractmethod
+    def get_option(self, option, hostvars=None):
+        pass
+
+    def _get_credentials(self):
```

Comment:
I would suggest to write this like that to avoid the loop:

```python
    def _get_credentials(self):
        '''
            :return A dictionary of boto client credentials
        '''
        boto_params = {
            'aws_access_key_id': self.aws_access_key_id,
            'aws_secret_access_key': self.aws_secret_access_key,
            'aws_session_token': self.aws_security_token,
        }
        return {k: v for k, v in boto_params.items() if v}
```